File: WK6/code/python/levene.py
import scipy.stats as stats
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt

df = pd.DataFrame()

# Levene's Test
statistic, p_value = stats.levene(
    df['Blood Glucose'][df['Diabetic Status'] == 'Diabetic'][df['Weight Status'] == 'Overweight'],
    df['Blood Glucose'][df['Diabetic Status'] == 'Diabetic'][df['Weight Status'] == 'Not Overweight'],
    df['Blood Glucose'][df['Diabetic Status'] == 'Non-diabetic'][df['Weight Status'] == 'Overweight'],
    df['Blood Glucose'][df['Diabetic Status'] == 'Non-diabetic'][df['Weight Status'] == 'Not Overweight']
)

print(f"Levenes Test P-value: {p_value} \n")


# Box Plot
plt.figure(figsize=(8, 6))
sns.boxplot(x=df['Diabetic Status'] + " & " + df['Weight Status'], y=df['Blood Glucose'])
plt.title("Box plot of Blood Glucose across Groups")
plt.xlabel("Groups")
plt.ylabel("Blood Glucose")
plt.xticks(rotation=45)
plt.show()

File: WK6/code/r/levene.r
install.packages("lawstat")
library(lawstat)

# Levene's Test
result <- levene.test(df$Blood_Glucose, df$Weight_Status)
print(result)


# OR Levene's Test
result <- levene.test(df$Blood_Glucose, interaction(df$Weight_Status, df$Blood_Glucose))
print(result)

library(ggplot2)

# Box Plot
ggplot(df, aes(x=interaction(Weight_Status), y=Blood_Glucose)) + 
  geom_boxplot() + 
  labs(title="Box plot of Blood Glucose across Groups", x="Groups", y="Blood Glucose")

